controllers/healthcheck_controller.go

```diff
@@ -170,68 +168,178 @@ func (r *HealthCheckReconciler) processHealthCheck(ctx context.Context, req ctrl
+		} else {
+			clusrole1, err := r.createRemedyClusterRole(r.kubeclient, amclusterRemedyRole)
+			if err != nil {
+				log.Error(err, "Error creating ClusterRole for the remedy workflow")
+			}
+			log.Info("Successfully Created", "ClusterRole", clusrole1)
+
+			crb1, err := r.createClusterRoleBinding(r.kubeclient, amclusterRoleRemedyBinding, amclusterRemedyRole, remedySa, wfNamespace)
 			if err != nil {
-				log.Error(err, "Error creating ServiceAccount for the workflow")
+				log.Error(err, "Error creating ClusterRoleBinding for the workflow")
 			}
-			log.Info("Successfully Created", "ServiceAccount", servacc)
+			log.Info("Successfully Created", "ClusterRoleBinding", crb1)
+
+		}
+
+	} else if level == "namespace" {
 
-			nsrole, err := r.CreateNameSpaceRole(r.kubeclient, amnsrole, wfNamespace)
+		if workFlowType != "remedy" {
+			nsRole, err := r.createNameSpaceRole(r.kubeclient, amnsRole, wfNamespace)
 			if err != nil {
 				log.Error(err, "Error creating NamespaceRole for the workflow")
 			}
-			log.Info("Successfully Created", "NamespaceRole", nsrole)
+			log.Info("Successfully Created", "NamespaceRole", nsRole)
 
-			nsrb, err := r.CreateNameSpaceRoleBinding(r.kubeclient, amnsrolebinding, nsrole, sa, wfNamespace)
+			nsrb, err := r.createNameSpaceRoleBinding(r.kubeclient, amnsRoleBinding, amnsRole, hcSa, wfNamespace)
 			if err != nil {
 				log.Error(err, "Error creating NamespaceRoleBinding for the workflow")
 			}
 			log.Info("Successfully Created", "NamespaceRoleBinding", nsrb)
 
 		} else {
-			return ctrl.Result{}, nil
+			nsRole1, err := r.createRemedyNameSpaceRole(r.kubeclient, amnsRemedyRole, wfNamespace)
+			if err != nil {
+				log.Error(err, "Error creating NamespaceRole for the workflow")
+			}
+			log.Info("Successfully Created", "NamespaceRole", nsRole1)
+
+			nsrb1, err := r.createNameSpaceRoleBinding(r.kubeclient, amnsRemedyRoleBinding, amnsRemedyRole, remedySa, wfNamespace)
+			if err != nil {
+				log.Error(err, "Error creating NamespaceRoleBinding for the workflow")
+			}
+			log.Info("Successfully Created", "NamespaceRoleBinding", nsrb1)
+
 		}
 
-		log.Info("Creating Workflow", "namespace", wfNamespace, "generateNamePrefix", wfNamePrefix)
-		generatedWfName, err := r.createSubmitWorkflow(ctx, log, healthCheck)
+	} else {
+		err := errors.New("level is not set")
+		return err
+	}
+
+	return nil
+}
+
+func (r *HealthCheckReconciler) deleteRBACForWorkflow(log logr.Logger, hc *activemonitorv1alpha1.HealthCheck) error {
+	level := hc.Spec.Level
+	remedySa := hc.Spec.RemedyWorkflow.Resource.ServiceAccount
+	wfNamespace := hc.Spec.RemedyWorkflow.Resource.Namespace
+
+	amclusterRemedyRole := remedySa + "-cluster-role"
+	amclusterRoleRemedyBinding := remedySa + "-cluster-role-binding"
+
+	amnsRemedyRole := remedySa + "-ns-role"
+	amnsRemedyRoleBinding := remedySa + "-ns-role-binding"
+	err := r.DeleteServiceAccount(r.kubeclient, remedySa, wfNamespace)
+	if err != nil {
```

Comment:
Let's look deeper into the error handling in this function. There are cases where even if there are errors, the execution simply moves ahead and is certainly going to crash the controller.

Reply:
sure.. will fix this to return error if any of the rbac methods fail either for deletion or creation